fetch.bs

```diff
@@ -2299,131 +2314,139 @@ the request.
+    <p>If <var>request</var> is a <a>subresource request</a>, run these substeps:
 
     <ol>
      <li>
-      <p>Set <var>value</var> to the first matching statement, if any, switching on
-      <var>hintName</var>:
-
-      <dl class=switch>
-       <dt>"<code>dpr</code>"
-       <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#dpr>dpr value</a>
-       <dt>"<code>save-data</code>"
-       <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#save-data>save-data value</a>
-       <dt>"<code>viewport-width</code>"
-       <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#viewport-width>viewport-width value</a>
-       <dt>"<code>width</code>"
-       <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#width>width value</a>
-      </dl>
-
-     <li><p><a for="header list">Append</a> <var>hintName</var>/<var>value</var> to
-     <var>request</var>'s <a for=request>header list</a>.
+      <p>If <var>request</var>'s <a for=client>client hints list</a> is not empty, then run these
+      substeps for each <var>hintName</var> in the list:
+
+      <ol>
+       <li>
+        <p>Set <var>value</var> to the first matching statement, if any, switching on
+        <var>hintName</var>:
+
+        <dl class=switch>
+         <dt>"<code>dpr</code>"
+         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#dpr>dpr value</a>
+         <dt>"<code>save-data</code>"
+         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#save-data>save-data value</a>
+         <dt>"<code>viewport-width</code>"
+         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#viewport-width>viewport-width value</a>
+         <dt>"<code>width</code>"
+         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#width>width value</a>
+        </dl>
+
+       <li><p><a for="header list">Append</a> <var>hintName</var>/<var>value</var> to
+       <var>request</var>'s <a for=request>header list</a>.
+      </ol>
+
+     <li><p>Let <var>record</var> be a new
+     <a for="fetch group">fetch record</a> consisting of
+     <var>request</var> and this instance of the
+     <a for=/>fetch</a> algorithm.
+
+     <li><p>Append <var>record</var> to <var>request</var>'s
+     <a for=request>client</a>'s
+     <a for=fetch>fetch group</a> list of
+     <a for="fetch group">fetch records</a>.
     </ol>
-
-   <li><p>Let <var>record</var> be a new
-   <a for="fetch group">fetch record</a> consisting of
-   <var>request</var> and this instance of the
-   <a for=/>fetch</a> algorithm.
-
-   <li><p>Append <var>record</var> to <var>request</var>'s
-   <a for=request>client</a>'s
-   <a for=fetch>fetch group</a> list of
-   <a for="fetch group">fetch records</a>.
   </ol>
 
+ <li><p>If fetch is <a for=fetch>terminated</a>, providing a <var>reason</var>, then return a
```

Comment:
I'm trying to avoid *reason* sounding like part of the conditional. As in "If the fetch is terminated, let the *reason* be the termination reason…" rather than "If the fetch is terminated and the termination reason is equal to *reason*"